### PR TITLE
Ens ccip wildcard support

### PR DESCRIPTION
**Pull request by makoto**
Relates to https://github.com/MetaMask/metamask-extension/pull/14675

NOTE: I was not able to setup iOs/Android env so the testing was limited to unit test.
For manual test, please use the same example records described at https://github.com/MetaMask/metamask-extension/pull/14675

### CODE REVIEW

**Comment by makoto** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by gantunesr — APPROVED** [review]
LGTM

Can you solve the conflicts before our QA team takes a look at it

**Comment by cortisiko** [comment]
@makoto friendly nudge. Please see the above comment

**Comment by makoto** [comment]
@mobularay sorry for missing the message. PTAL

**Comment by gantunesr** [comment]
Hey @makoto, can you run `yarn install` and `yarn update` in your branch? Looks like the yarn.lock file is not right

**Comment by makoto** [comment]
@gantunesr is this ok now?

**Comment by makoto** [comment]
FYI, like the metamask extension PR, please use 1.offchainexample.eth to check it it resolve to Ethereum addresses

**Comment by mobularay** [comment]
Awesome @makoto, really appreciate your responsiveness. I'll let either @gantunesr or @cortisiko confirm the next steps.

To set expectations: internally, we have a Personal Development day tomorrow and then an internal ConsenSys Productivity Zero day on Monday. Our QA team may not have enough time to do their due diligence by our RC cutoff time on Tuesday morning. The team will do their best; if this ticket doesn't make it for v5.5 RC cutoff on Tuesday, then this will definitely get included in our next release v5.6 in August. Sorry if this is TMI, hope this makes sense. Thank you!

File: app/util/ENSUtils.test.ts [hunk 1]
```diff
@@ -1,24 +1,53 @@
-import { isDefaultAccountName } from './ENSUtils';
+import { isDefaultAccountName, getEnsProvider } from './ENSUtils';
+import sinon from 'sinon';
```
Comment (gantunesr) [inline app/util/ENSUtils.test.ts:2]: I think it's not necessary to use `sinon` here, you can just use Jest mocks

**Comment by gantunesr** [comment]
I see an error in the test workflow @makoto 

`ReferenceError: You are trying to import a file after the Jest environment has been torn down.`

**Comment by makoto** [comment]
> I see an error in the test workflow @makoto
> 
> `ReferenceError: You are trying to import a file after the Jest environment has been torn down.`

hi @gantunesr could you let me know where I can find this error?

**Comment by mobularay** [comment]
Hi @makoto, my team is not working today July 25 - see explanation [here](https://github.com/MetaMask/metamask-mobile/pull/4322#issuecomment-1191745917). Everyone will be back in action tomorrow July 26, and we'll follow-up with you on resolving the merge conflict. There is still the feature QA process that this ticket is required to go through, so we will make sure to get this in for the next release in August (v5.6). 

We'll be in touch!

File: app/util/ENSUtils.test.ts [hunk 2]
```diff
@@ -1,24 +1,49 @@
...
```
Comment (gantunesr) [inline app/util/ENSUtils.test.ts:2]: Remove `sinon`

**Comment by gantunesr** [comment]
@makoto you can run `yarn test` to get a trace stack of the failed test

**Comment by makoto** [comment]
@gantunesr the `ReferenceError: You are trying to import a file after the Jest environment has been torn down.
` error seems happening even on the latest main branch so I don't think it's related to my PR

```
metamask/metamask-mobile - (main) > git log | head -5
commit a0549ac35cdc11a436ab0e0db0facba00f240f1f
Author: Cal Leung <cleun007@gmail.com>
Date:   Fri Jul 29 13:14:12 2022 -0700

    Use default QR code colors (#4639)
metamask/metamask-mobile - (main) > npx jest app/components/Views/WalletConnectSessions/index.test.tsx
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating

ReferenceError: You are trying to `import` a file after the Jest environment has been torn down.



      at Object.get StyleSheet [as StyleSheet] (node_modules/react-native/index.js:387:12)
      at createStyles (app/components/Views/WalletConnectSessions/index.js:6171:23)
      at WalletConnectSessions._this.render (app/components/Views/WalletConnectSessions/index.js:6720:34)
      at ReactShallowRenderer._updateClassComponent (node_modules/enzyme-adapter-react-16/node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:948:39)
      at ReactShallowRenderer.render (node_modules/enzyme-adapter-react-16/node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:781:12)
 PASS  app/components/Views/WalletConnectSessions/index.test.tsx (7.575 s)
  WalletConnectSessions
    ✓ should render correctly (6 ms)

  console.warn
    PushNotificationIOS has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-community/push-notification-ios' instead of 'react-native'. See https://github.com/react-native-push-notification-ios/push-notification-ios

      at warnOnce (node_modules/react-native/Libraries/Utilities/warnOnce.js:27:11)
      at Object.get PushNotificationIOS [as PushNotificationIOS] (node_modules/react-native/index.js:372:5)
      at Object.<anonymous> (node_modules/react-native-push-notification/component/index.ios.js:10:27)


=============================== Coverage summary ===============================
Statements   : 25.03% ( 2114/8445 )
Branches     : 12.37% ( 607/4906 )
Functions    : 13.15% ( 196/1490 )
Lines        : 26.7% ( 1087/4071 )
================================================================================
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   1 passed, 1 total
Time:        8.064 s
Ran all test suites matching /app\/components\/Views\/WalletConnectSessions\/index.test.tsx/i.
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
metamask/metamask-mobile - (main) > 
```

**Comment by gantunesr** [comment]
You're right, all good @makoto. We can move to QA now @cortisiko @plasmacorral @mobularay

**Comment by cortisiko** [comment]
@makoto Unfortunately I am unable to find the ENS name for **1.offchainexample.eth** 

I get "No address has been set for this name" See [here](http://recordit.co/rbHDCTAGrm)

cc @mobularay @gantunesr

**Comment by makoto** [comment]
As mentioned in the original PR, I wasn't able to live debug as I wasn't able to build the mobile app on the iOS emulator. Is there a way to do UI debugging without launching iOS emulator?

**Comment by gantunesr** [comment]
This code has been merged into a feature branch and its not included in a release yet

**Pull request merged**